src/api.cpp

```diff
@@ -392,6 +398,48 @@ std::string handle_entity(std::string command, std::string args, uchar *command_
     return "";
 }
 
+std::string handle_pollChatPosts(std::string command, std::string args, uchar *command_server) {
+    if (chat_log.size() == 0) {
+        return "\n";
+    }
+
+    std::string history = "";
+
+    for (const std::string& message : chat_log) {
+        history += message;
+        // Use a null byte instead of pipes so I don't have to deal with escapping
+        history.push_back('\0');
+    }
+
+    chat_log.clear();
+    history.pop_back();
+    return history + "\n";
+}
+
+std::string handle_setChatLogSize(std::string command, std::string args, uchar *command_server) {
+    try {
+        chat_log_size = std::stoi(args);
+        return "1\n";
+    } catch (const std::invalid_argument& e) {
+        return "\n";
+    }
+}
+
+std::string handle_getVersion(std::string command, std::string args, uchar *command_server) {
+    return std::string(*minecraft_pi_version) + "\n";
+}
+
+HOOK(chat_send_message, void, (unsigned char *server_side_network_handler, char *username, char *message)) {
```

Comment:
This works on servers/the hosting player, but this function isn't called on clients.

Reply:
I don't know of another easy-to-hook function that doesn't require some form of parsing. :/

Reply:
`Gui::addMessage` should be used, but currently can't (without disabling Reborn stuff) until a bug with `extract_from_bl_instruction` is fixed (or a workaround is found).